sdk/helper/scaleutils/filter.go

```diff
@@ -88,37 +99,78 @@ const nodeAttrGCEZone = "platform.gce.zone"
 const defaultClassIdentifier = "autoscaler-default-pool"
 
 // filterByClass returns a filtered list of nodes which are active in the
-// cluster and where the specified class matches that of the nodes.
-func filterByClass(n []*api.NodeListStub, id string) []*api.NodeListStub {
+// cluster and where the specified class matches that of the nodes. In the
+// event that nodes are found within the class pool in an unstable state, and
+// thus indicating there is change occurring; an error will be returned.
+func filterByClass(n []*api.NodeListStub, id string) ([]*api.NodeListStub, error) {
 
 	// Create our output list object.
 	var out []*api.NodeListStub
 
+	// Track all nodes which are deemed to cause an error to the autoscaler. It
+	// is possible to make an argument that the first error should be returned
+	// in order to improve speed. In a situation where two nodes are in an
+	// undesired state, it would require the operator to perform the same tidy
+	// and restart of the autoscaler loop twice, which seems worse than having
+	// some extra time within this function.
+	var err *multierror.Error
+
 	for _, node := range n {
 
-		// Ignore nodes that are not in a ready state.
-		if node.Status != api.NodeStatusReady {
+		// Filter out all nodes which do not match the target class first.
+		if node.NodeClass != "" && node.NodeClass != id ||
+			node.NodeClass == "" && id != defaultClassIdentifier {
 			continue
 		}
 
-		// Ignore nodes that are not currently eligible.
-		if node.SchedulingEligibility != api.NodeSchedulingEligible {
+		// We should class an initializing node as an error, this is caused by
+		// node registration and could be sourced from scaling out.
```

Comment:
should we be nervous here? is there a guarantee in nomad that a client will eventually move from init to ready? otherwise, this could block cluster scaling forever.